### PR TITLE
Feature/GH-242 reveal password

### PR DESCRIPTION
**Pull request by MoaazThahabee**
Create and use reveal password feature

### CODE REVIEW

File: SafeUIKit/SafeUIKit/Components/TextInputs/VerifiableInput/VerifiableInput.swift [hunk 1]
```diff
@@ -220,6 +241,13 @@ open class VerifiableInput: UIView {
     @objc private func textChanged(_ sender: Any) {
         text = textInput.text // validation
     }
+    
+    @objc private func toggleRevealPassword(_ sender: Any) {
+        guard let togglePasswordButton = self.togglePasswordButton else {return}
```
Comment (sche) [inline SafeUIKit/SafeUIKit/Components/TextInputs/VerifiableInput/VerifiableInput.swift]: Please use spaces
```
{ return }
```

**Comment by sche** [comment]
👍

File: SafeUIKit/SafeUIKit/Components/TextInputs/VerifiableInput/VerifiableInput.swift [hunk 2]
```diff
@@ -108,6 +109,26 @@ open class VerifiableInput: UIView {
...
+    public var isPassword: Bool = false {
+        didSet {
+            isSecure = isPassword
+            if isPassword {
+                let button = UIButton(type: .custom)
+                button.frame = TextInput.normalAccessoryRect
+                button.addTarget(self, action: #selector(toggleRevealPassword), for: .touchUpInside)
+                button.setImage(Asset.passwordRevealIcon.image, for: .normal)
+                button.setImage(Asset.passwordHideIcon.image, for: .selected)
+                self.rigthView = button
```
Comment (DmitryBespalov) [inline SafeUIKit/SafeUIKit/Components/TextInputs/VerifiableInput/VerifiableInput.swift]: To be consistent with the project's style, the variables outside of closures should omit `self`.

File: SafeUIKit/SafeUIKit/Components/TextInputs/VerifiableInput/VerifiableInput.swift [hunk 3]
```diff
@@ -220,6 +241,13 @@ open class VerifiableInput: UIView {
     @objc private func textChanged(_ sender: Any) {
         text = textInput.text // validation
     }
+    
+    @objc private func toggleRevealPassword(_ sender: Any) {
+        guard let togglePasswordButton = self.togglePasswordButton else {return}
+        
+        self.isSecure = !self.isSecure
```
Comment (DmitryBespalov) [inline SafeUIKit/SafeUIKit/Components/TextInputs/VerifiableInput/VerifiableInput.swift]: here, too, `self`s are not needed

**Review by DmitryBespalov — CHANGES_REQUESTED** [review]
Good job! A few more steps and we're done here.

**Review by DmitryBespalov — APPROVED** [review]

**Pull request merged**